src/sgl/device/device.cpp

```diff
@@ -91,6 +91,96 @@ Device::Device(const DeviceDesc& desc)
         std::filesystem::create_directories(m_shader_cache_path);
     }
 
+    // Invalidate CUDA interop if using CUDA
+    if (m_desc.type == DeviceType::cuda && m_desc.enable_cuda_interop) {
+        m_desc.enable_cuda_interop = false;
+        log_warn("Device type is set to CUDA, but CUDA interop is requested. enable_cuda_interop will be ignored.");
```

Comment:
Should this be a warning? Nothing unexpected would happen if you request CUDA interop on a CUDA device that the user should be made aware of, right? It's just that the flag is not needed.